### PR TITLE
page object pattern

### PR DESCRIPTION
**Pull request by olaleks**
Screen Shot 2022-05-21 at 3 02 20 PM

### CODE REVIEW

File: src/main/java/com/stormnet/page/ContactUsPage.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        return this;
+    }
+
+    public ContactUsPage selectOrderNumber(String orderNum){
+        selectorOrderNumber.selectOptionContainingText(orderNum);
+        return this;
+    }
+
+    public ContactUsPage selectOrderItem(){
+        $("select[name=\"id_product\"][id=\"" + getOrderId() + "_order_products\"]").selectOption(1);
```
Comment (Kirill-Simonovich) [inline src/main/java/com/stormnet/page/ContactUsPage.java]: use string format

File: src/main/java/com/stormnet/page/ReviewPage.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    return this;
+  }
+
+  public ReviewPage setCommentTitle() {
+    commentTitle.sendKeys("High quality product");
+    return this;
+  }
+
+  public ReviewPage setReviewContent() {
+    reviewContent.sendKeys(
```
Comment (Kirill-Simonovich) [inline src/main/java/com/stormnet/page/ReviewPage.java:24]: Why always send this message, what if I want to send another message?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 3]
```diff
@@ -1,188 +1,140 @@
...
+            .clickProcessCarrierBtn()
+            .clickPaymentWithBankWire()
+            .clickConfirmOrder();
+    String orderNumber = confirmedOrderPage.getOrderNumber();
+    confirmedOrderPage
+            .clickContactUs()
+            .selectSubjectHeading()
+            .selectOrderNumber(orderNumber)
+            .selectOrderItem()
+            .clickSighOutButton();
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: Where are the checks?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 4]
```diff
@@ -1,188 +1,140 @@
...
+            .hoverFirstDress()
+            .clickAddToCartBtn()
+            .clickProceedToCheckoutBtn()
+            .clickProceedToCheckoutBtn()
+            .clickProceedAddressBtn()
+            .selectAgreeTermsCheckBox()
+            .clickProcessCarrierBtn()
+            .clickPaymentWithBankWire()
+            .clickConfirmOrder()
+            .clickSighOutButton();
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: Where are the checks(Assertions)?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 5]
```diff
@@ -1,188 +1,140 @@
...
+            .clickTshirtMenu()
+            .hoverTshirt()
+            .clickViewButton()
+            .clickWriteReview()
+            .setStarMark()
+            .setCommentTitle()
+            .setReviewContent()
+            .clickSubmitReview()
+            .clickSubmitBtn()
+            .clickSighOutButton();
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: Where are the checks(Assertions)?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 6]
```diff
@@ -1,188 +1,140 @@
...
+            .clickViewButton()
+            .clickWishlistButton()
+            .shouldHaveAlertWindow()
+            .clickCloseBtn()
+            .clickAccountButton()
+            .clickMyWishlistButton()
+            .clickMyWishlist()
+            .clickWishlistDelete()
+            .acceptAlert()
+            .clickSighOutButton();
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: Where are the checks(Assertions)?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 7]
```diff
@@ -1,188 +1,140 @@
...
+            .login("test54321@yopmail.com", "12345")
+            .hoverDressingMenu()
+            .clickSummerDressesSubMenu()
+            .hoverFirstSummerDress()
+            .clickViewButton()
+            .clickBlackColor()
+            .clickOrangeColor()
+            .clickBlueColor()
+            .clickYellowColor()
+            .clickSighOutButton();
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: Where are the checks(Assertions)?

File: src/test/java/com/stormnet/TestCasesAutomation.java [hunk 8]
```diff
@@ -1,188 +1,140 @@
...
-    $("#id_state").selectOptionByValue("9");
-    $("#postcode").sendKeys("33131");
-    $("#id_country").selectOptionByValue("21");
-    $("#other").sendKeys("Additional Information");
-    $("#phone").sendKeys("+375292929297");
-    $("#phone_mobile").sendKeys("+375333333333");
-    $("#alias").sendKeys("18, Baker street");
-    $("#submitAccount").click();
-    $(".logout[title=\"Log me out\"]").click();
+    HomePage homePage = open("/", HomePage.class);
```
Comment (Kirill-Simonovich) [inline src/test/java/com/stormnet/TestCasesAutomation.java]: take it out in the beforeeeah method

**Review by Kirill-Simonovich — CHANGES_REQUESTED** [review]

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 8 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed
Screen Shot 2022-05-21 at 3 02 20 PM
d

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 7 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 6 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 5 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 4 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed

File: src/test/java/com/stormnet/TestCasesAutomation.java (hunk 3 above)
Comment (olaleks) [reply src/test/java/com/stormnet/TestCasesAutomation.java]: fixed

File: src/main/java/com/stormnet/page/ReviewPage.java (hunk 2 above)
Comment (olaleks) [reply src/main/java/com/stormnet/page/ReviewPage.java:24]: fixed

File: src/main/java/com/stormnet/page/ContactUsPage.java (hunk 1 above)
Comment (olaleks) [reply src/main/java/com/stormnet/page/ContactUsPage.java]: fixed

Comment (Kirill-Simonovich) [reply src/main/java/com/stormnet/page/ContactUsPage.java]: private String orderItem= "select[name=\"id_product\"][id=\"%s_order_products\"]";

$(String.format(orderItem, getOrderId())).selectOption(1);

**Review by Kirill-Simonovich — APPROVED** [review]

**Pull request merged**